File: frontend/src/types/share.ts
export interface ShareLinkEntity {
  id: number
  userId: number
  fileId: number
  shareCode: string
  hasPassword: boolean
  expireAt: string | null
  maxVisits: number | null
  downloadCount: number
  visitCount: number
  isCanceled: boolean
  createdAt: string
  updatedAt: string
}

export interface SharePasswordResponse {
  password: string | null
}

export interface CreateShareRequest {
  fileId: number
  expireAt?: number
  maxVisits?: number
  password?: string
  notifyEmail?: string
}

export interface UpdateShareRequest {
  expireAt?: number
  maxVisits?: number
  password?: string
}

export interface AccessShareResponse {
  fileId: number
  fileName: string
  fileType: 'file' | 'folder'
  fileSize: number
}

File: desktop-client/src/betterclouddrive_desktop/models/share.py
from __future__ import annotations

from typing import Literal

from .api import AppModel


class ShareLinkEntity(AppModel):
    id: int
    userId: int
    fileId: int
    shareCode: str
    passwordHash: str | None = None
    expireAt: str | None = None
    maxVisits: int | None = None
    downloadCount: int = 0
    visitCount: int = 0
    isCanceled: bool = False
    createdAt: str = ""
    updatedAt: str = ""


class AccessShareResponse(AppModel):
    fileId: int
    fileName: str
    fileType: Literal["file", "folder"]
    fileSize: int = 0

